### PR TITLE
Tickets/dm-14908: Convert package to numpydoc format

### PR DESCRIPTION
**Pull request by isullivan**

### CODE REVIEW

File: python/lsst/obs/test/dualRawImage.py [hunk 1]
```diff
@@ -20,6 +20,8 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 
+__all__ = ("RawAndFlatContainer", "RawAndFlatAssembler", "RawAndFlatDisassembler")
```
Comment (isullivan) [inline python/lsst/obs/test/dualRawImage.py]: It looks like the other packages use a list for `__all__` , so I would recommend a list here for consistency.

File: python/lsst/obs/test/makeTestRawVisitInfo.py [hunk 2]
```diff
@@ -19,29 +19,40 @@
...
-        """Set an argument dict for VisitInfo and pop associated metadata
+        """Set an argument dict for VisitInfo and pop associated metadata.
 
-        @param[in,out] md  metadata, as an lsst.daf.base.PropertyList or PropertySet
-        @param[in,out] argdict  a dict of arguments
+        Parameters
+        ----------
+        md : `lsst.daf.base.PropertySet`
+            Image metadata.
+        `dict`
```
Comment (isullivan) [inline python/lsst/obs/test/makeTestRawVisitInfo.py]: Missing `argDict`?

File: python/lsst/obs/test/testMapper.py [hunk 3]
```diff
@@ -66,8 +67,15 @@ def _extractDetectorName(self, dataId):
...
-        @param dataId (dict) Dataset identifier
-        @return (string) path to the defects file or None if not available
+
+        Parameters
+        ----------
+        dataId : `dict`
+            Dataset identifier
+        Returns
+        -------
+        result : str
```
Comment (isullivan) [inline python/lsst/obs/test/testMapper.py]: `str`

File: python/lsst/obs/test/makeTestRawVisitInfo.py [hunk 4]
```diff
@@ -19,29 +19,40 @@
...
+            A dict of arguments for the `lsst.afw.image.VisitInfo`
+            constructor. Updated by this call.
+
+        Returns
+        -------
+        `lsst.afw.image.VisitInfo`
```
Comment (isullivan) [inline python/lsst/obs/test/makeTestRawVisitInfo.py]: Here and below, include a name for the return.

Comment (r-owen) [reply python/lsst/obs/test/makeTestRawVisitInfo.py]: OK. That used to be optional but I guess it isn't any longer.

**Pull request merged**